specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/dataContainer.json

```diff
@@ -0,0 +1,351 @@
+              "schema": {
+                "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+              }
+            }
+          }
+        },
+        "put": {
+          "tags": [
+            "DataContainer"
+          ],
+          "x-ms-examples": {
+            "Create or update code container": {
+              "$ref": "./examples/createOrUpdateDataContainer.json"
+            }
+          },
+          "description": "Create or update Data Container.",
+          "operationId": "DataContainers_CreateOrUpdate",
+          "parameters": [
+            {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+            },
+            {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+            },
+            {
+              "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+            },
+            {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+            },
+            {
+              "$ref": "#/parameters/DataContainerNameParameter"
+            },
+            {
+              "in": "body",
+              "name": "body",
+              "description": "Data Version entity to create or update.",
+              "schema": {
+                "$ref": "#/definitions/AssetContainerResourceRequest"
+              }
+            }
+          ],
+          "responses": {
+            "200": {
+              "description": "Success",
+              "schema": {
+                "$ref": "#/definitions/AssetContainerResource"
+              }
+            },
+            "default": {
+              "description": "Error",
+              "schema": {
+                "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+              }
+            }
+          }
+        }
+      }
+    },
+    "parameters": {
+      "DataContainerNameParameter": {
+        "in": "path",
+        "name": "name",
+        "type": "string",
+        "description": "Data Container identifier.",
+        "required": true,
+        "x-ms-parameter-location": "method"
+      }
+    },
+    "definitions": {
+      "AssetContainerResource": {
+        "type": "object",
+        "properties": {
+          "id": {
+            "type": "string",
+            "description": "The resource URL of the entity (not URL encoded).",
+            "readOnly": true
+          },
+          "name": {
+            "type": "string",
+            "description": "The name of the resource entity.",
+            "readOnly": true
+          },
+          "type": {
+            "type": "string",
+            "description": "The resource provider and type.",
+            "readOnly": true
+          },
+          "properties": {
+            "$ref": "#/definitions/DataContainer"
```

Comment:
>DataContainer [](start = 35, length = 13)

Where is this defined?

Reply:
please take a look at the updated swagger; this is now AssetContainer